ethereum/messages.py

```diff
@@ -238,7 +235,8 @@ def _apply_casper_no_gas_transaction(state, tx):
     if not result:
         log_tx.debug('TX FAILED', reason='out of gas',
                      startgas=tx.startgas, gas_remained=gas_remained)
-        raise InvalidCasperVote("Failed Casper vote is considered an invalid tx")
+        output = b''
+        success = 0
```

Comment:
Need to return here instead of continuing to execute the rest of the function. Can use
```python
return 0, b''
```

Reply:
It's okay ultimately because the failed return raises an `assert`, but we should just get out of the function sooner rather than later

Reply:
or maybe just move assert statement here? So it doesn't make the validity of early exit looks too implicit. And also make all casper no gas tx logics under same function?
```
if not result:
    log_tx.debug('TX FAILED', reason='out of gas',
                     startgas=tx.startgas, gas_remained=gas_remained)
    assert False, "CASPER no gas transaction should always success"
```